src/ln/channel.rs

```diff
@@ -2135,6 +2135,19 @@ impl Channel {
 		}
 		Channel::check_remote_fee(fee_estimator, msg.feerate_per_kw)?;
 
+		//Confirm sender can afford the new fee rate when applied to the current local commitment transaction
+		let num_htlcs = if self.last_local_commitment_txn.len() > 0 {
+			self.last_local_commitment_txn[0].output.len()
+		} else {
+			0
+		};
+
+		let total_fee: u64 = msg.feerate_per_kw as u64 * (COMMITMENT_TX_BASE_WEIGHT + (num_htlcs as u64) * COMMITMENT_TX_WEIGHT_PER_HTLC) / 1000;
```

Comment:
Hmm, the spec says we "MAY delay this check until the update_fee is committed" which may make this much easier to reason about - in other words instead of doing this check in update_fee do it in commitment_signed by looking at the resulting commitment transaction that is generated and make sure that the output to our counterparty didn't get pruned.

Reply:
I have moved the check out to the commitment_signed message as it does make more sense there. Had to be careful to only check it if the fee was updated.